### PR TITLE
[GJ-115]Update rename to Gluten in source code.

### PR DESCRIPTION
**Pull request by weiting-chen**
Closes https://github.com/oap-project/gluten/issues/115.

### CODE REVIEW

File: docs/memory.md [hunk 1]
```diff
@@ -8,12 +8,12 @@ Modern memory allocators like jemalloc will not return the just freed memory to
 
 ## Turnings to reduce memory footprint
 
-- Gazelle Plugin 1.2+
+- Gluten Plugin 1.2+
```
Comment (QiangCai) [inline docs/memory.md:11]: the version should be 1.0+

**Comment by zzcclp** [comment]
Verified with ClickHouse Backend, LGTM.

**Comment by rui-mo** [comment]
@marin-ma please help review and test this PR, thanks.

**Comment by zzcclp** [comment]
Verified with ClickHouse Backend

**Review by zzcclp — APPROVED** [review]

**Comment by marin-ma** [comment]
Verified with Arrow backend, LGTM.

**Comment by rui-mo** [comment]
Verified locally. LGTM.

**Pull request merged**